### PR TITLE
Picturae import

### PR DESCRIPTION
**Pull request by delarocamcas**
First round of changes to update importer to newer data standards:
-picturae_csv_create: removed and split off into new repo 
-associated unittests with picturae_csv_create removed as well
-picturae_importer:  removed locality_id pull, to make updates to localities from notes from nature easier, a new locality ID will be created for every new collection object.
-gen_import_utils: in order to maintain parallel changes with utils in picturae_batch_wrangler repo, 2 new functions were added , picturae_paths_lists function changed to reflect new folder structure.
-img_batch_resizer: function run from command line in order to  create a compressed and resized jpeg copy of tiff batch folders . Used to resize image batches to notes from nature standards.

### CODE REVIEW

File: PIC_dbcreate/Picturae_DDL.sql
Comment (delarocamcas) [inline PIC_dbcreate/Picturae_DDL.sql:1]: removing taxa unmatch table: TNRS flagged taxa will now be dealt with at the front end.

File: client_tools.py
Comment (delarocamcas) [inline client_tools.py:1]: added a separate analogous updater to the Picturae updater, in order to perform routine updates on non-picturae, botany specify data if needed.

File: gen_import_utils.py [hunk 1]
```diff
@@ -5,13 +5,61 @@
 import sys
 import numpy as np
 import pandas as pd
-import json
 import hmac
 import settings
 import os
-
+from string_utils import remove_non_numerics
 # import list tools
```
Comment (delarocamcas) [inline gen_import_utils.py:13]: used to seperate out titles from name columns

File: gen_import_utils.py [hunk 2]
```diff
@@ -5,13 +5,61 @@
...
 
+
```
Comment (delarocamcas) [inline gen_import_utils.py:14]: used to separate titles from name columns

File: gen_import_utils.py [hunk 3]
```diff
@@ -5,13 +5,61 @@
...
+    # Check and process last name titles
+    for title in last_name_titles:
+        if row['Last Name'].endswith(" " + title):
+            row['Title'] = title
+            row['Last Name'] = row['Last Name'][:-(len(title)+1)]
+            break
+
+    return row
+
+
```
Comment (delarocamcas) [inline gen_import_utils.py:47]: used to combine date columns into one properly formatted date column according to casbotany db standards. Used in picturae_batch_wrangler, but added here to have parallel changes across util files.

File: gen_import_utils.py [hunk 4]
```diff
@@ -27,26 +75,6 @@ def unique_ordered_list(input_list):
     return unique_elements
```
Comment (delarocamcas) [inline gen_import_utils.py:77]: removed, as no longer using json format configs

File: gen_import_utils.py [hunk 5]
```diff
@@ -55,19 +83,15 @@ def extract_last_folders(path, number: int):
```
Comment (delarocamcas) [inline gen_import_utils.py:83]: no longer pulling images from nested date paths, so modified to use simple date input and config prefix.

File: img_batch_resizer.py
Comment (delarocamcas) [inline img_batch_resizer.py:1]: used to create copies of tiff folders in compressed jpeg format. Created to populate notes from nature folders with higher quality jpegs for transcription (the picturae supplied jpegs were low quality, made handwriting hard to read). Let me know if you think this should live somewhere else.

File: picturae_importer.py [hunk 6]
```diff
@@ -8,18 +8,16 @@
...
 from sql_csv_utils import SqlCsvTools
 from botany_importer import BotanyImporter
-from picturae_csv_create import starting_time_stamp
-from specify_db import SpecifyDb
 from get_configs import get_config
 from os import path
 import time_utils
 import logging.handlers
 from monitoring_tools_derived import MonitoringToolsDir
 from taxon_tools.BOT_TNRS import process_taxon_resolve
```
Comment (delarocamcas) [inline picturae_importer.py:19]: moved starting timestamp to beginning of importer rather than picturae_csv_create

File: picturae_importer.py [hunk 7]
```diff
@@ -123,7 +111,7 @@ def run_timestamps(self, batch_size: int):
                         AND TimestampCreated <= "{ending_time_stamp}";'''
```
Comment (delarocamcas) [inline picturae_importer.py:113]: removed taxa_unmatch insert statement as no longer needed.

File: picturae_importer.py [hunk 8]
```diff
@@ -865,8 +850,7 @@ def upload_records(self):
             if self.taxon_id is None:
                 self.create_taxon()
```
Comment (delarocamcas) [inline picturae_importer.py:852]: now creating new locality ID for every new collection object, to allow for smoother updating when notes from nature comes back.

File: sql_csv_utils.py [hunk 9]
```diff
@@ -267,6 +265,8 @@ def create_update_statement(self, tab_name: str, col_list: list, val_list: list,
                 val_list: list of values with which to update above list of columns(order matters)
                 condition: condition sql string used to select sub-sect of records to update.
         """
+        val_list, col_list = remove_two_index(value_list=val_list, column_list=col_list)
+
```
Comment (delarocamcas) [inline sql_csv_utils.py:269]: protection against NAs in sql statement

File: specify_db.py
Comment (delarocamcas) [inline specify_db.py:1]: removed erroneous testing print statements

File: sql_csv_utils.py [hunk 10]
```diff
@@ -284,28 +284,6 @@ def create_update_statement(self, tab_name: str, col_list: list, val_list: list,
 
         return sql
```
Comment (delarocamcas) [inline sql_csv_utils.py:286]: no longer useful sql_csv_tools method, no longer using a taxa_unmatch table to track TNRS flagged taxa.

File: img_batch_resizer.py
Comment (foozleface) [reply img_batch_resizer.py:1]: This looks like it should be standalone? or at least in a subdir.

**Review by foozleface — APPROVED** [review]
do the breakout in the next commit, I think.

**Pull request merged**